### PR TITLE
Eek

### PR DESCRIPTION
**Pull request by MangoIV**
- added eek shield 
- added eek basic keymap
- didn't change every commit since begin of the project :D

### CODE REVIEW

**Review by Nicell — CHANGES_REQUESTED** [review]
Thanks for adding support for this shield! All the DT stuff looks pretty solid.

There's a couple files that still need to be updated:


[`.github/workflows/build.yml`](https://github.com/zmkfirmware/zmk/blob/main/.github/workflows/build.yml): Add your board to the shield list.

[`docs/docs/hardware.md`](https://github.com/zmkfirmware/zmk/blob/main/docs/docs/hardware.md): You should add the eek! to this list

[`docs/static/setup.ps1`](https://github.com/zmkfirmware/zmk/blob/main/docs/static/setup.ps1) and [`docs/static/setup.sh`](https://github.com/zmkfirmware/zmk/blob/main/docs/static/setup.sh): eek! should be added as a non-split option for both files.

**Comment by MangoIV** [comment]
Will do those tomorrow. Thanks.

**Comment by MangoIV** [comment]
added the eek! to the requested files. Thanks for reviewing!

**Review by Nicell — APPROVED** [review]
Great! Thanks.

File: app/boards/shields/eek/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,6 @@
+if SHIELD_EEK
```
Comment (Nicell) [inline app/boards/shields/eek/Kconfig.defconfig:1]: Could you add a license header here?

File: app/boards/shields/eek/Kconfig.shield [hunk 2]
```diff
@@ -0,0 +1,2 @@
+config SHIELD_EEK
```
Comment (Nicell) [inline app/boards/shields/eek/Kconfig.shield:1]: License header here, too, please! 😄

**Review by Nicell — CHANGES_REQUESTED** [review]
Missed a couple license header spots. @MangoIV once these are fixed up I think this is good to go.

**Comment by MangoIV** [comment]
I just add "Copyright blabla " to every newly created file, yes?

**Comment by Nicell** [comment]
> I just add "Copyright blabla " to every newly created file, yes?

@MangoIV specifically the two I mentioned in the review, but in general, yes. The `readme` for example doesn't need it though.

**Comment by MangoIV** [comment]
should be fine now :)

**Comment by MangoIV** [comment]
sorry for the rough process, will be easier for the next shields :D

**Review by Nicell — APPROVED** [review]

File: app/boards/shields/eek/eek.overlay [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ */
+
+#include <dt-bindings/zmk/matrix_transform.h>
+
+/ {
+    chosen {
+        zmk,kscan = &kscan0;
+        zmk,matrix_transform = &default_transform;
+    };
+
```
Comment (innovaker) [inline app/boards/shields/eek/eek.overlay:14]:
```suggestion
```

File: app/boards/shields/eek/eek.overlay [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            , <&pro_micro_a 2 GPIO_ACTIVE_HIGH>
+            , <&pro_micro_a 3 GPIO_ACTIVE_HIGH>
+            ;
+
+        row-gpios
+            = <&pro_micro_d 6 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            , <&pro_micro_d 7 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            , <&pro_micro_d 8 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            , <&pro_micro_d 9 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+            ;
```
Comment (innovaker) [inline app/boards/shields/eek/eek.overlay:51]: Very minor, but I'd be remiss if I didn't mention that this style of formatting strays from the unwritten conventions generally found in existing shields.  And yes, we really need a standard formatting tool.

File: docs/docs/hardware.md [hunk 5]
```diff
@@ -36,6 +36,7 @@ That being said, there are currently only a few specific [boards](/docs/faq#what
 - [QAZ](https://www.cbkbd.com/product/qaz-keyboard-kit) (`qaz`)
 - [CRBN](https://keygem.store/collections/group-buys/products/group-buy-featherlight-40-kit) (`crbn`)
 - [tidbit](https://nullbits.co/tidbit/) (`tidbit`)
+- [eek!](https://www.cbkbd.com/product/eek-keyboard) (`eek`)
```
Comment (innovaker) [inline docs/docs/hardware.md:39]: You added a `!` to the keyboard name setup descriptions.  Should this link also follow that style?

Also, I noticed you used a capital "Eek!" in setup and "eek!" in the keyboard name.

File: app/boards/shields/eek/eek.overlay [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    RC(1,9) RC(1,8) RC(1,7) RC(1,6) RC(1,5) RC(1,4) RC(1,3) RC(1,2) RC(1,1) RC(1,0)
+    RC(2,9) RC(2,8) RC(2,7) RC(2,6) RC(2,5) RC(2,4) RC(2,3) RC(2,2) RC(2,1) RC(2,0)
+                    RC(3,7) RC(3,6) RC(3,5) RC(3,4) RC(3,3) RC(3,2) 
+            >;
+    };
+
+    kscan0: kscan_0 {
+    compatible = "zmk,kscan-gpio-matrix";
+    label = "KSCAN";
+    diode-direction = "col2row";
```
Comment (innovaker) [inline app/boards/shields/eek/eek.overlay:31]: The indentation here looks off?

**Review by innovaker — COMMENTED** [review]
Thanks!  Very minor suggestions.

File: app/boards/shields/eek/eek.overlay (hunk 4 above)
Comment (MangoIV) [reply app/boards/shields/eek/eek.overlay:51]: I don't see the problem here, tbh

Comment (innovaker) [reply app/boards/shields/eek/eek.overlay:51]: The whole block has an unnecessary extra indentation.

Comment (innovaker) [reply app/boards/shields/eek/eek.overlay:51]: You know what, it was the lack of indentation in the lines above throwing off my spidey sense.  My bad.

**Comment by MangoIV** [comment]
Should be fine now?

**Review by innovaker — APPROVED** [review]

**Pull request merged**